packages/back-end/src/services/features.ts

```diff
@@ -81,6 +82,30 @@ export function getFeatureDefinition({
     rules:
       rules
         ?.filter((r) => r.enabled)
+        ?.filter((r) => {
+          //Filter our rules that have schedulingRules if the current scheduleRule sets the feature to disabled.
+          if (!r.scheduleRules?.length) {
+            return true;
+          }
+
+          const currentDate = new Date();
+
+          // Loop through the list of rules, and find the next rule to be applied
+          const nextRuleIndex = r.scheduleRules.findIndex(
+            (rule) =>
+              rule.timestamp !== null && new Date(rule.timestamp) > currentDate
+          );
+
+          if (nextRuleIndex === -1) {
+            if (r.scheduleRules.at(-1)?.timestamp === null) {
+              return true;
+            } else {
+              return r.scheduleRules.at(-1)?.enableFeature;
+            }
+          }
+
+          return !r.scheduleRules[nextRuleIndex].enableFeature;
```

Comment:
This logic doesn't feel quite right.  What if the scheduledRules array is not sorted?  What if there is only 1 rule in the future that says "enable"?

This logic should live in it's own util function and have unit tests that cover all of the edge cases.  Even if the code above does work and covers all the edge cases, there's no way to know just by looking at it.

```ts
getCurrentEnabledState(r.scheduledRules, currentDate)
```